pipeline/types/engine.py

```diff
@@ -54,3 +54,10 @@ class MatchingOutput:
     users: List[User]
     matches: List[Match]
     internal_data: MatchingInternalData = MatchingInternalData()
+
+
+@dataclass
+class MatchingInternalAnalytics:  # More metrics will be added in the future
```

Comment:
Now that it's not internal anymore, we can rename this to `MatchingMetrics`, I think metrics is a better description than analytics.